lib/LedgerSMB/Template/DB.pm

```diff
@@ -131,13 +132,16 @@ does not call this without carefully whitelisting values.
 
 sub get_from_file {
     my ($package, $path, $language_code) = @_;
-    my $fname = $path;
+
     if ($path =~ m|/.*:| ){
        die 'Cannot run on NTFS alternate data stream!';
     }
-    $path =~ m|(.*)/([^/]+)$|;
-    $fname = $2;
+
+    my ( undef, undef, $fname)  = File::Spec->splitpath( $path);
+    die "Bad template file name $fname, should have one dot."
```

Comment:
Just one remark: the split being unequal to two could mean either fewer (1) or more (3+) values being returned. Elsewhere in the code I used `split(<regex>, <value>, 2)` to restrict to return at most 2. Is that sane here too?